libvast/src/system/spawn.cpp

```diff
@@ -172,6 +173,26 @@ expected<actor> spawn_metastore(local_actor* self, options& opts) {
   return actor_cast<actor>(s);
 }
 
+expected<actor> spawn_metastore_simple(local_actor* self,
+                                       [[maybe_unused]] options& opts) {
+  auto store = self->spawn(simple_store<std::string, data>,
+                           opts.dir / "simple_store");
+  return actor_cast<actor>(store);
+}
+
+expected<actor> spawn_metastore(local_actor* self, options& opts) {
+  auto impl = "simple"s;
+  auto r = opts.params.extract_opts({
+    {"impl", "the store backend (simple|raft)", impl},
+  });
+  opts.params = r.remainder;
+  if (impl == "simple")
+    return spawn_metastore_raft(self, opts);
```

Comment:
`spawn_metastore_raft` => `spawn_metastore`?

Reply:
I actually like the suffix indicating the concrete implementation type. Then `spawn_metastore` does the dispatching to the concrete functions.

Reply:
But then `spawn_metastore` would have to evaluate the `impl` argument. Or did you mean we should also rename `spawn_metastore` to `spawn_metastore_simple`?